api_tests/lib/actor_test.ts

```diff
@@ -1,69 +1,62 @@
 import { Actors } from "./actors";
 import { createActors } from "./create_actors";
-import { timeout } from "./utils";
+import { HarnessGlobal, timeout } from "./utils";
+import JasmineSmacker from "smack-my-jasmine-up";
+
+declare var global: HarnessGlobal;
 
 /*
  * This test function will take care of instantiating the actors and tearing them down again
  * after the test, regardless if the test succeeded or failed.
  */
-function nActorTest(
-    name: string,
+async function nActorTest(
     actorNames: ["alice", "bob", "charlie"] | ["alice", "bob"] | ["alice"],
     testFn: (actors: Actors) => Promise<void>
 ) {
+    const name = JasmineSmacker.getCurrentTestName();
     if (!name.match(/[A-z0-9\-]+/)) {
         // We use the test name as a file name for the log and hence need to restrict it.
         throw new Error(
             `Testname '${name}' is invalid. Only A-z, 0-9 and dashes are allowed.`
         );
     }
 
-    it(name, async function() {
-        this.timeout(100_000); // absurd timeout. we have our own one further down
-        const actors = await createActors(`${name}.log`, actorNames);
+    const actors = await createActors(`${name}`, actorNames);
 
-        try {
-            await timeout(60000, testFn(actors));
-        } catch (e) {
-            for (const actorName of actorNames) {
-                await actors.getActorByName(actorName).dumpState();
-            }
-            throw e;
-        } finally {
-            for (const actorName of actorNames) {
-                actors.getActorByName(actorName).stop();
-            }
+    try {
+        await timeout(60000, testFn(actors));
+    } catch (e) {
+        for (const actorName of actorNames) {
+            await actors.getActorByName(actorName).dumpState();
+        }
+        throw e;
+    } finally {
+        for (const actorName of actorNames) {
+            actors.getActorByName(actorName).stop();
         }
-    });
+    }
 }
 
 /*
- * Instantiates a new e2e test based on three actors
- *
+ * Instantiates a new e2e test based on one actor
  */
-export function threeActorTest(
-    name: string,
-    testFn: (actors: Actors) => Promise<void>
-) {
-    nActorTest(name, ["alice", "bob", "charlie"], testFn);
+export async function oneActorTest(testFn: (actors: Actors) => Promise<void>) {
```

Comment:
Please do the re-ordering to another PR. This PR is big enough, this can of changes just makes review harder.